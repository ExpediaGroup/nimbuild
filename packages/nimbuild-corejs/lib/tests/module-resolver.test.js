const {getModules} = require('../module-resolver');
const {getSupported, getBaseFeatureModules} = require('../supported-sets');
const mockuas = require('../mocks/ua.mock');

describe('module-resolver.js', () => {
    let include;
    let exclude;
    let mockLogger; //eslint-disable-line

    beforeEach(() => {
        mockLogger = {
            log: jest.fn()
        };
        const featureSet = getSupported('default', {
            log: console.log
        });
        include = featureSet.include;
        exclude = featureSet.exclude;
    });

    it('Resolves chrome UA to list of coreJS modules', async () => {
        const features = getBaseFeatureModules({
            include,
            exclude
        });
        const response = await getModules({
            features,
            uaString: mockuas.chrome,
            minify: false,
            logger: mockLogger
        });

        expect(response.corejs.length).toEqual(0);
        expect(response.normal.length).toEqual(0);
        expect(response).toMatchSnapshot();
    });

    it('Resolves ie11 UA to list of coreJS modules', async () => {
        const features = getBaseFeatureModules({
            include,
            exclude
        });
        const response = await getModules({
            features,
            uaString: mockuas.ie11,
            minify: false,
            logger: mockLogger
        });

        expect(response.corejs.length).toEqual(110);
        expect(response.normal.length).toEqual(2);
        expect(response).toMatchSnapshot();
    });
});
