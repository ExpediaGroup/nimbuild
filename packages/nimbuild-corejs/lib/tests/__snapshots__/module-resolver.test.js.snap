// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module-resolver.js Resolves chrome UA to list of coreJS modules 1`] = `
Object {
  "corejs": Array [],
  "normal": Array [],
}
`;

exports[`module-resolver.js Resolves future versions of chrome to modern bundle 1`] = `
Object {
  "corejs": Array [],
  "normal": Array [],
}
`;

exports[`module-resolver.js Resolves ie11 UA to list of coreJS modules 1`] = `
Object {
  "corejs": Array [
    "es.symbol",
    "es.symbol.description",
    "es.symbol.async-iterator",
    "es.symbol.has-instance",
    "es.symbol.is-concat-spreadable",
    "es.symbol.iterator",
    "es.symbol.match",
    "es.symbol.match-all",
    "es.symbol.replace",
    "es.symbol.search",
    "es.symbol.species",
    "es.symbol.split",
    "es.symbol.to-primitive",
    "es.symbol.to-string-tag",
    "es.symbol.unscopables",
    "es.array.concat",
    "es.array.copy-within",
    "es.array.every",
    "es.array.fill",
    "es.array.filter",
    "es.array.find",
    "es.array.find-index",
    "es.array.flat",
    "es.array.flat-map",
    "es.array.for-each",
    "es.array.from",
    "es.array.includes",
    "es.array.index-of",
    "es.array.iterator",
    "es.array.join",
    "es.array.last-index-of",
    "es.array.map",
    "es.array.of",
    "es.array.reduce",
    "es.array.reduce-right",
    "es.array.slice",
    "es.array.some",
    "es.array.species",
    "es.array.splice",
    "es.array.unscopables.flat",
    "es.array.unscopables.flat-map",
    "es.date.to-primitive",
    "es.function.has-instance",
    "es.function.name",
    "es.json.stringify",
    "es.json.to-string-tag",
    "es.map",
    "es.number.constructor",
    "es.number.epsilon",
    "es.number.is-finite",
    "es.number.is-integer",
    "es.number.is-nan",
    "es.number.is-safe-integer",
    "es.number.max-safe-integer",
    "es.number.min-safe-integer",
    "es.number.parse-float",
    "es.number.parse-int",
    "es.number.to-fixed",
    "es.object.assign",
    "es.object.define-getter",
    "es.object.define-setter",
    "es.object.entries",
    "es.object.freeze",
    "es.object.from-entries",
    "es.object.get-own-property-descriptor",
    "es.object.get-own-property-descriptors",
    "es.object.get-own-property-names",
    "es.object.get-prototype-of",
    "es.object.is",
    "es.object.is-extensible",
    "es.object.is-frozen",
    "es.object.is-sealed",
    "es.object.keys",
    "es.object.lookup-getter",
    "es.object.lookup-setter",
    "es.object.prevent-extensions",
    "es.object.seal",
    "es.object.to-string",
    "es.object.values",
    "es.promise",
    "es.promise.all-settled",
    "es.promise.finally",
    "es.regexp.constructor",
    "es.regexp.exec",
    "es.regexp.flags",
    "es.regexp.sticky",
    "es.regexp.test",
    "es.regexp.to-string",
    "es.set",
    "es.string.code-point-at",
    "es.string.ends-with",
    "es.string.from-code-point",
    "es.string.includes",
    "es.string.iterator",
    "es.string.match",
    "es.string.match-all",
    "es.string.pad-end",
    "es.string.pad-start",
    "es.string.repeat",
    "es.string.replace",
    "es.string.search",
    "es.string.split",
    "es.string.starts-with",
    "es.string.trim",
    "es.string.trim-end",
    "es.string.trim-start",
    "web.dom-collections.for-each",
    "web.dom-collections.iterator",
    "web.queue-microtask",
    "web.url-search-params",
  ],
  "normal": Array [
    "whatwg-fetch",
    "intersection-observer",
    "custom-event-polyfill",
  ],
}
`;
